main.rb

```diff
@@ -0,0 +1,159 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    arr = to_a
+    while i <= arr.length - 1
+      yield(arr[i])
+      i += 1
+    end
+  end
+
+  def all_validate
+    arr = self
+    if block_given? == false && (arr.include?(false) == true || arr.include?(nil) == true)
+      false
+    elsif block_given? == false && (arr.include?(false) == false && arr.include?(nil) == false)
+      true
+    end
+  end
+
+  def my_all?(*)
+    arr = to_a
+    if block_given? == false
+      if arr.all_validate == false
+        false
+      elsif arr.all_validate == true
+        true
+      end
+    else
+      arr = to_a
+      count = 0
+      arr.my_each do |x|
+        count += 1 if yield(x)
+      end
+      count == arr.length
+    end
+  end
```

Comment:
- [ ] when a class is passed as an argument returns false if any of the collection is not a member of such class => You need to handle if the class is passed as an argument (you can use if statements)

- [ ] when a Regex is passed as an argument returns false if any of the collection does not match the Regex => Same as the class, but for Regex

- [ ]  when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern => Same of the above, but with patterns.